File: backend/Customer/Customer/Models/MealmenuFooditem.cs
using System;
using System.Collections.Generic;

namespace Customer.Models;

public partial class MealmenuFooditem
{
    public int MealmenuFooditemId { get; set; }

    public int MenuId { get; set; }

    public int FoodId { get; set; }

    public virtual FoodItem Food { get; set; } = null!;

    public virtual MealMenu Menu { get; set; } = null!;
}

File: backend/MessOwner/src/main/java/com/example/entities/MealMenuFoodItem.java
package com.example.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "mealmenu_fooditem")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class MealMenuFoodItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "mealmenu_fooditem_id")
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "menu_id")
    @JsonIgnore
    private MealMenu mealMenu;

    @ManyToOne
    @JoinColumn(name = "food_id")
    private FoodItem foodItem;
}
